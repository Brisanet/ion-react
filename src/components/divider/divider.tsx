import { Divider, DividerContainer } from './styled';

export type DividerDirection = 'horizontal' | 'vertical';
export type DividerType = 'solid' | 'dashed';

export type DividerProps = {
  label?: string;
  direction?: DividerDirection;
  type?: DividerType;
};

export const IonDivider = ({
  label,
  direction = 'horizontal',
  type = 'solid',
}: DividerProps) => {
  return label ? (
    <DividerContainer>
      <Divider $direction='horizontal' $type={type} />
      <span>{label}</span>
      <Divider $direction='horizontal' $type={type} />
    </DividerContainer>
  ) : (
    <Divider $direction={direction} $type={type} />
  );
};
