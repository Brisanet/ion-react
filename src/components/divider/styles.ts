import styled, { RuleSet, css } from 'styled-components';
import { DividerDirection, DividerType } from './divider';

type DividerProps = {
  $direction: DividerDirection;
  $type: DividerType;
};

const type: (type: DividerType) => RuleSet<object> = (type) => {
  return {
    solid: css`
      border-style: solid;
    `,
    dashed: css`
      border-style: dashed;
    `,
  }[type];
};

const directions: (direction: DividerDirection) => RuleSet<object> = (
  direction
) => {
  return {
    horizontal: css`
      border-top: 1px;
      width: 100%;
    `,
    vertical: css`
      border-left: 1px;
      height: 100%;
      width: 0;
    `,
  }[direction];
};

export const DividerContainer = styled.div`
  ${({ theme }) => css`
    ${theme.utils.flex.start(8)}

    & span {
      color: ${theme.colors.neutral[7]};
      ${theme.font.size[14]}
    }
  `}
`;

export const Divider = styled.hr<DividerProps>`
  ${({ theme, $direction, $type }) => css`
    border-width: 0px;
    ${directions($direction)}
    ${type($type)}
    border-color: ${theme.colors.neutral[4]};
  `}
`;
