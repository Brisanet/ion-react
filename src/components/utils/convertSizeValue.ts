/**
 * Get a size value based on its type.
 * @param {string | number} value - The size value.
 * @returns The value itself if it's a string, or the value in pixels if it is a number.
 **/
export const convertSizeValue = (value: string | number): string =>
  typeof value === 'number' ? `${value}px` : value;
