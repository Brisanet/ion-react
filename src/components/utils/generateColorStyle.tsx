// Tipo para objeto de estilos  da função geradora de estilos
type ColorStyle = {
    background: string;
    color: string;
    border?: string;
  };
  
  /* 
    Função que gera cores de Background, Border e cor de
    fonte a fim de reduzir repetições de codigo
  
    Seguir o padrão de parâmetros: 
    1- Cor do background;
    2- Cor da fonte do botão;
    3- (Opicional) Cor da borda;
  */
 export function generateColorStyle(
    bgColor: string,
    fontColor: string,
    borderColor: string
  ) {
    let styles: ColorStyle = {
      background: bgColor,
      color: fontColor,
    };
  
    borderColor && (styles.border = `1px solid ${borderColor}`);
  
    return styles;
  }