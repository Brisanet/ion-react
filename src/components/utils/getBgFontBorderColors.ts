interface ObjStyles {
  background: string;
  color: string;
  border?: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param background - Background color
 * @param color - The font color
 * @param border - Border attribute styles - example: ´1px solid blue´
 * @returns A Javascript object containing color properties.
 */
export function getBgFontBorderColors({
  background,
  color,
  border = '',
}: ObjStyles): ObjStyles {
  return {
    background,
    color,
    border,
  };
}
