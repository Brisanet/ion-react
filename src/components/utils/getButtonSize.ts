/**
 * Function to prevent repetition of style blocks.
 *
 * @param buttonSize - Height and min-width
 * @param padding - Button padding
 * @param fontSize - The size of font
 * @returns An Javascript object containing size properties.
 */
export function getButtonSize(
  buttonSize: string,
  padding: string,
  fontSize: string
) {
  let size = {
    height: buttonSize,
    minWidth: buttonSize,
    padding: padding,
    fontSize: fontSize,
  };

  return size;
}
