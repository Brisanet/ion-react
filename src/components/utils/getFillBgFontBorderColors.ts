type ColorStyleResponse = {
  border: string;
  backgroundColor: string;
  color: string;
  svg: {
    fill: string;
  };
};

interface StyleProps {
  svgFillColor: string;
  backgroundColor: string;
  fontColor: string;
  borderColor: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param svgFillColor - Fill color to be used in SVG
 * @param backgroundColor - Background color
 * @param fontColor - The font color
 * @param borderColor - Border color
 * @returns A Javascript object containing style properties.
 */
export function getFillBgFontBorderColors({
  backgroundColor,
  fontColor,
  borderColor,
  svgFillColor,
}: StyleProps) {
  let styles: ColorStyleResponse = {
    backgroundColor: backgroundColor,
    color: fontColor,
    border: `1px solid ${borderColor}`,
    svg: {
      fill: svgFillColor,
    },
  };

  return styles;
}
