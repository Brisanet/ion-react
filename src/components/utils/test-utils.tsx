import theme from '@ion/styles/theme';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
