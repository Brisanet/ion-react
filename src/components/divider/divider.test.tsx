import { screen } from '@testing-library/react';
import { renderWithTheme } from '../utils/test-utils';
import { DividerProps, IonDivider } from './divider';

const sut = (props?: DividerProps) => {
  return renderWithTheme(<IonDivider {...props} />);
};

const directions: Array<DividerProps['direction']> = ['horizontal', 'vertical'];
const types: Array<DividerProps['type']> = ['dashed', 'solid'];
const label = 'Label';

describe('Divider', () => {
  describe('Default', () => {
    it.each(directions)('should render %s divider', (direction) => {
      const { container } = sut({ direction });
      expect(container).toMatchSnapshot();
    });

    it.each(types)('should render %s divider', (type) => {
      const { container } = sut({ type });
      expect(container).toMatchSnapshot();
    });
  });

  describe('With Label', () => {
    it('should render with a label', () => {
      sut({ label });
      expect(screen.getByText(label)).toBeVisible();
    });
  });
});
