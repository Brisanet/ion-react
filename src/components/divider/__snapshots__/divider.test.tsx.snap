// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Default should render dashed divider 1`] = `
.c0 {
  border-width: 0px;
  border-top: 1px;
  width: 100%;
  border-style: dashed;
  border-color: #ced2db;
}

<div>
  <hr
    class="c0"
  />
</div>
`;

exports[`Divider Default should render horizontal divider 1`] = `
.c0 {
  border-width: 0px;
  border-top: 1px;
  width: 100%;
  border-style: solid;
  border-color: #ced2db;
}

<div>
  <hr
    class="c0"
  />
</div>
`;

exports[`Divider Default should render solid divider 1`] = `
.c0 {
  border-width: 0px;
  border-top: 1px;
  width: 100%;
  border-style: solid;
  border-color: #ced2db;
}

<div>
  <hr
    class="c0"
  />
</div>
`;

exports[`Divider Default should render vertical divider 1`] = `
.c0 {
  border-width: 0px;
  border-left: 1px;
  height: 100%;
  width: 0;
  border-style: solid;
  border-color: #ced2db;
}

<div>
  <hr
    class="c0"
  />
</div>
`;
